lib/tools/xctest-commands.js

```diff
@@ -0,0 +1,51 @@
+import os from 'os';
+
+const DELIMETER = ' | ';
+
+const xctestCommands = {};
+
+/**
+ * Runs an xcuitest on the device or simulator
+ * @param {string} testRunnerBundleId The bundle id of the test runner
+ * @param {string} appUnderTestBundleId The bundle id of the app under test
+ * @param {string} xctestBundleId The bundle id of the xctest package
+ * @returns {SubProcess}
+ */
+xctestCommands.runXCUITest = async function runXCUITest (testRunnerBundleId, appUnderTestBundleId, xctestBundleId) {
+  const uiTestProcess = this.createSubProcess([
+    'xctest', 'run', 'ui', '--udid', this.udid, xctestBundleId, appUnderTestBundleId, testRunnerBundleId
+  ]);
+  await uiTestProcess.start(0);
```

Comment:
should we check if there was a failure while starting the process?

Reply:
What would be the error btw? I'm not fully familiar with the api

Reply:
this depends on the actual process output. We could set a startup detector function if it prints some particular strings to the output. Or just way for a while and check if the process still alive like we do in https://github.com/appium/appium-idb/blob/6ab51a092c00d6bcd6ac94541186e3f8499eeebb/lib/tools/system-commands.js#L94

Reply:
I checked the command it seems like it is not printing anything. We will be able to check if the WDA is up via the /status endpoint